crates/swc_common/src/source_map.rs

```diff
@@ -1222,6 +1223,16 @@ impl SourceMap {
                 continue;
             }
 
+            if source_root.is_none() {
+                if let FileName::Real(f) = &f.name {
+                    if let Some(parent) = f.parent() {
+                        let s = parent.display().to_string();
+                        builder.set_source_root(Some(&s));
```

Comment:
This looks like you're pulling `sourceRoot` from the the first file that's iterated in mappings, but I think that's going to give a lot of bugs.

Do injected code helper have a synthetic source file in the mappings? Or, can SWC generate a single output from multiple inputs (Babel has a `{ 'file1.js': '...', 'file2.js': '...' }` transformer).

Further, `sourceRoot` is generally configured via options. Eg, [Babel](https://babeljs.io/docs/en/options#sourceroot), [Terser](https://github.com/terser/terser#:~:text=You%20can%20also%20specify%20sourceRoot%20property%20to%20be%20included%20in%20source%20map%3A), [Closure](https://github.com/google/closure-compiler/wiki/Flags-and-Options#:~:text=source_map_location_mapping), [esbuild](https://esbuild.github.io/api/#source-root)

Reply:
Oh, thank you!
I'll check them

Reply:
Injected helpers are skipped, but this can be problematic for bundler